app/src/androidTest/java/chat/rocket/android/chatroom/ui/ChatRoomFragmentTest.kt

```diff
@@ -1,67 +1,164 @@
+        onView(withId(R.id.image_avatar)).check(matches(isDisplayed()))
+        onView(withId(R.id.text_sender)).check(matches(isDisplayed()))
+        onView(withId(R.id.text_message_time)).check(matches(isDisplayed()))
+        onView(withId(R.id.text_content)).check(matches(isDisplayed()))
+        onView(withId(R.id.text_content)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun show_attachment_options() {
+        navigateToExistingChannel()
+        onView(withId(R.id.button_show_attachment_options)).perform(click())
+        onView(withId(R.id.button_take_a_photo)).check(matches(isDisplayed()))
+        onView(withId(R.id.button_attach_a_file)).check(matches(isDisplayed()))
+        onView(withId(R.id.button_drawing)).check(matches(isDisplayed()))
+        onView(withId(R.id.text_message)).check(matches(isDisplayed()))
+    }
 
-        fragment.showFileSelection(null)
+    @Test
+    fun send_text_message() {
+        navigateToExistingChannel()
+        onView(withId(R.id.text_message)).check(matches(withHint(R.string.msg_message)))
+            .perform(
+            typeText(TEST_MESSAGE), closeSoftKeyboard()
+        )
+        onView(withId(R.id.button_send)).check(matches(isDisplayed()))
+            .perform(click())
+        onView(withId(R.id.text_message)).perform(clearText())
+    }
 
-        intended(
-            allOf(
-                hasAction(Intent.ACTION_GET_CONTENT),
-                hasType("*/*"),
-                hasCategories(setOf(Intent.CATEGORY_OPENABLE)),
-                not(hasExtraWithKey(Intent.EXTRA_MIME_TYPES))
+    @Test
+    fun check_message_action_bottom_sheet() {
+        navigateToExistingUser2()
+        onView(withId(R.id.recycler_view)).perform(
+            RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(
+                0, click()
             )
         )
+        onView(withText(R.string.action_msg_add_reaction)).check(matches(isDisplayed()))
+        onView(withText(R.string.action_msg_reply)).check(matches(isDisplayed()))
+        onView(withText(R.string.action_msg_quote)).check(matches(isDisplayed()))
+        onView(withText(R.string.action_msg_permalink)).check(matches(isDisplayed()))
+        onView(withText(R.string.action_msg_copy)).check(matches(isDisplayed()))
+        onView(withText(R.string.action_msg_edit)).check(matches(isDisplayed()))
+        onView(withText(R.string.action_info)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun clicking_user_avatar_should_open_his_details(){
+        navigateToExistingUser2()
+        onView(withId(R.id.recycler_view)).perform(ScrollToTop())
+        onView(withId(R.id.recycler_view)).perform(
+            RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(
+                0, clickChildViewWithId(R.id.image_avatar)
+            )
+        )
+        onView(withId(R.id.user_details_layout)).check(matches(isDisplayed()))
+    }
+
+    private fun loginIfUserIsLoggedOut() {
```

Comment:
@GOVINDDIXIT This function appears in many files.....  is it possible to move it to a single file which can be imported by the tests?  If there are any other "utility" functions like this maybe you can start building a little library file where they can live....

Reply:
Sure @ear-dev I will move it to a separate class with other utility functions. Thanks 👍